cachi2/core/package_managers/yarn/resolver.py

```diff
@@ -344,13 +343,44 @@ def _generate_purl_for_package(package: _ResolvedPackage, project: Project) -> s
     :param project: the project object to resolve the configured registry url and file paths
         for file dependencies.
     """
-    # registry url can be accessed in project.yarnrc
-    # paths for file dependencies are relative to project.source_dir
+    qualifiers = dict()
+    subpath = None
+
+    # package with NpmLocator doesn't need any qualifiers
+
+    if isinstance(package.locator, HttpsLocator):
+        qualifiers["download_url"] = package.locator.url
+        if package.checksum is not None:
+            qualifiers["checksum"] = f"sha512-{package.checksum}"
```

Comment:
The separator between algorithm and checksum should be a `:`. 

https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#generic